### PR TITLE
Check for Deprecated on method page

### PR DESCRIPTION
**Pull request by kstole**
#48 updated the spiders to grab methods from JSON data included on the [Web API methods](https://api.slack.com/methods) page. As noted in [this comment](https://github.com/slack-ruby/slack-api-ref/issues/47#issuecomment-989076232), the deprecation info in that data can be inaccurate.

This change continues using that deprecation data but also checks for an individual method reference page labeling the method deprecated (as is the case for `rtm.start`).

### CODE REVIEW

**Comment by kstole** [comment]
@dblock It’s been quite a while since I contributed to the Slack gems. I just haven’t been working on Slack bots the last few years. But I recently had to patch our company’s bot to work with `rtm.connect` so I’m drafting a few PRs to update that in the main gems as well. I’m also participating in [Hacktoberfest](https://hacktoberfest.com/participation/) so it would be great if you can add the `hacktoberfest-accepted` label on those PRs and maybe `hacktoberfest` on the overall repos could encourage some participation. Thanks!

**Comment by dblock** [comment]
Thanks @kstole! Want to help me maintain this repo? Then you can create some hacktoberfest labels :)

**Review by dblock — CHANGES_REQUESTED** [review]
I don't think this is right. We collect deprecated methods and mark them `deprecated: true` in the JSON. This skips them?

File: tasks/lib/slack_api/methods_spider.rb [hunk 1]
```diff
@@ -36,7 +36,9 @@ def process_list(page, _default_data = {})
 
     def process_method(page, default_data = {})
       method_page = ensure!(page, '.apiMethodPage', default_data[:method_name])
-      desc = method_page.search('.apiReference__mainDescription').text.gsub("’", "'")
+      desc = method_page.search('.apiReference__mainDescription').text.gsub('’', "'")
+      return if desc.downcase.start_with? 'deprecated:'
```
Comment (dblock) [inline tasks/lib/slack_api/methods_spider.rb:40]: Shouldn't this change the value of `deprecated: ` below instead?

**Comment by dblock** [comment]
Consider separating the `"` changes into a separate PR that would enable rubocop and do a pass with that on the repo code with `-a`.

File: tasks/lib/slack_api/methods_spider.rb (hunk 1 above)
Comment (kstole) [reply tasks/lib/slack_api/methods_spider.rb:40]: It looks like that used to be the behavior but was changed in #48 to hardcode `'deprecated' => false` and just remove any deprecated methods, so this is an extra check for when `isDeprecated` isn't accurate
https://github.com/slack-ruby/slack-api-ref/blob/f213e377393aed5c466832f087d40cd35d07274e/tasks/lib/slack_api/methods_spider.rb#L15

**Comment by kstole** [comment]
> Thanks @kstole! Want to help me maintain this repo? Then you can create some hacktoberfest labels :)

Yeah for sure. Happy to help maintain

**Comment by dblock** [comment]
> > Thanks @kstole! Want to help me maintain this repo? Then you can create some hacktoberfest labels :)
> 
> Yeah for sure. Happy to help maintain

Added you, https://github.com/slack-ruby/slack-api-ref/invitations. Please continue making PRs, but feel free to merge others' and make labels as you please.

**Review by dblock — APPROVED** [review]

**Pull request merged**